setup.py

```diff
@@ -29,6 +34,7 @@ def local_pkg(name: str, relative_path: str) -> str:
     "h5netcdf",  # for xarray
     "dask",  # for xarray
     "numpy==1.26.4",
+    "matplotlib",  # for plotting in boilerplate
```

Comment:
`ndsl/boilerplate.py`  is used in a bunch of places, e.g. in unit tests. I thus couldn't add `matplotlib` only for the `[demos]` flavor.

Reply:
Not great, but we were heading there anyway